yaptide/celery/utils/pymc.py

```diff
@@ -112,23 +112,29 @@ def __call__(self, file_path: str, rng_seed: int) -> None:
     update_fluka_function(str(input_file.resolve()), random_seed)
 
 
-def execute_simulation_subprocess(dir_path: Path, command_as_list: list[str]) -> tuple[bool, str, str]:
+def execute_simulation_subprocess(dir_path: Path,
+                                  command_as_list: list[str],
+                                  task_id: int,
+                                  update_key: str = '',
+                                  simulation_id: int = Optional[None]) -> tuple[bool, str, str]:
     """Function to execute simulation subprocess."""
     process_exit_success: bool = True
     command_stdout: str = ""
     command_stderr: str = ""
     try:
-        completed_process = subprocess.run(command_as_list,
-                                           check=True,
-                                           cwd=str(dir_path),
-                                           stdout=subprocess.PIPE,
-                                           stderr=subprocess.PIPE,
-                                           text=True)
-        logging.info("simulation subprocess with return code %d finished", completed_process.returncode)
-
+        process = subprocess.Popen(command_as_list,
+                                   cwd=str(dir_path),
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE,
+                                   text=True)
+        pid = process.pid
+        logging.info("sending info to task: %d in simulation: %d with update key: %s to update PID %d", task_id,
```

Comment:
why `simulation_id or -1` feeded into `%d` ??

Reply:
simulation_id is set to None is tests (thts why some tests failing before, they failed on this log). I will wroite it in comment

Reply:
do you know why `simulation_id` is set to none ? that seems like sign of some bug.
Hiding bug in complex logic is a bad habit, better to report it in an issue.
In the issue you can write that you suspect sim_id being None and you should say how to reproduce it.

In the code here I would rather use somehow more verbose code, like:

```
if simulation_id:
   logging.info(...)
   send_task_update(...)
else:
  # we may have simulation_id being none, reported in an issue ISSUE_URL...
  logging.info(....)
```

does it make sense to update (send task update) a task which doesn't belong to any simulation ? do we have such tasks ?

Reply:
Then please write it down in an issue and put here the link in the coment